pkg/controller/elasticsearch/driver/downscale.go

```diff
@@ -86,11 +86,11 @@ func calculateDownscales(expectedStatefulSets sset.StatefulSetList, actualStatef
 
 // scheduleDataMigrations requests Elasticsearch to migrate data away from leavingNodes.
 // If leavingNodes is empty, it clears any existing settings.
-func scheduleDataMigrations(esClient esclient.Client, leavingNodes []string) error {
+func scheduleDataMigrations(allocationSetter esclient.AllocationSetter, leavingNodes []string) error {
 	if len(leavingNodes) != 0 {
 		log.V(1).Info("Migrating data away from nodes", "nodes", leavingNodes)
 	}
-	return migration.MigrateData(esClient, leavingNodes)
+	return migration.MigrateData(allocationSetter, leavingNodes)
```

Comment:
I realize this isn't part of your change, but I wonder if we still want to keep this func around since it is essentially just a wrapper for `migration.MigrateData`